src/components/OTPInput.vue

```diff
@@ -0,0 +1,199 @@
+
+import { useLoaderStore } from '@/stores/loader.store'
+import useArcanaAuth from '@/use/arcanaAuth'
+
+const AppOverlay = defineAsyncComponent(
+  () => import('@/components/AppOverlay.vue')
+)
+
+const emit = defineEmits<{
+  (e: 'dismiss'): void
+  (e: 'resend'): void
+  (e: 'success'): void
+}>()
+
+const otpLength = 6
+const attemptsUsed = ref(0)
+const totalAttemptsAllowed = 3
+const resendCounter = ref(30)
+
+const otp = ref('')
+const otpInputs = ref<HTMLInputElement[]>([])
+const arcanaAuth = useArcanaAuth()
+const loader = useLoaderStore()
+const hasOTPError = ref(false)
+startOTPCounterTimer()
+
+function startOTPCounterTimer() {
+  resendCounter.value = 30
+  const interval = setInterval(() => {
+    --resendCounter.value
+    if (resendCounter.value === 0) {
+      clearInterval(interval)
+    }
+  }, 1000)
+}
+
+function handlePaste(event: ClipboardEvent) {
+  hasOTPError.value = false
+  if (!event.clipboardData) return
+  const paste = event.clipboardData.getData('text/plain')
+  if (paste.startsWith('0x') || isNaN(Number(paste))) return
+  if (paste.length === otpLength) {
+    otp.value = paste
+  }
+}
+
+function focusInput(index: number) {
+  otpInputs.value[index]?.focus()
+}
+
+const isValidOTP = computed(
+  () => otp.value.length === otpLength && !isNaN(Number(otp.value))
+)
+
+function handleKeyDown(event: KeyboardEvent, index: number) {
+  hasOTPError.value = false
+  if ([event.code, event.key].includes('Backspace')) {
+    event.preventDefault()
+    otp.value = otp.value.slice(0, index) + otp.value.slice(index + 1)
+    focusInput(index - 1)
+  } else if (event.code === 'Delete') {
+    event.preventDefault()
+    otp.value = otp.value.slice(0, index) + otp.value.slice(index + 1)
+  } else if (event.code === 'ArrowLeft') {
+    event.preventDefault()
+    if (index !== 0) focusInput(index - 1)
+  } else if (event.code === 'ArrowRight') {
+    event.preventDefault()
+    if (index !== 5) focusInput(index + 1)
+  } else if (
+    event.code === 'Spacebar' ||
+    event.code === 'Space' ||
+    event.code === 'ArrowUp' ||
+    event.code === 'ArrowDown'
+  ) {
+    event.preventDefault()
+  } else if (!isNaN(Number(event.key))) {
+    event.preventDefault()
+    otp.value =
+      otp.value.slice(0, index) + event.key + otp.value.slice(index + 1)
+    focusInput(index + 1)
+  }
+}
+
+async function submitOTP() {
+  hasOTPError.value = false
+  try {
+    loader.showLoader('Verifying the OTP...')
+    await arcanaAuth.getAuthInstance().loginWithOTPComplete(otp.value)
+    emit('success')
+    emit('dismiss')
+  } catch (error) {
+    console.error(error)
+    hasOTPError.value = true
+  } finally {
+    loader.hideLoader()
+  }
+}
+
+async function resendOTP() {
+  emit('resend')
+  startOTPCounterTimer()
+}
+</script>
+
+<template>
+  <AppOverlay>
+    <div
+      class="max-w-[360px] w-screen bg-[#1e1f1e] rounded-[10px] border-1 border-jet flex flex-col relative p-[1.75rem] gap-4"
```

Comment:
Add all custom styles to tailwind config theme extend and use it here.

`p-[1.75rem]` = `p-7`